internal/controller/progressive/progressive.go

```diff
@@ -274,12 +279,6 @@ func processUpgradingChild(
 
 		// if so, mark the existing one for garbage collection and then create a new upgrading one
 		if needsUpdating {
-			if liveRolloutObject.GetPromotedChildStatus() == nil {
-				rolloutObject.SetPromotedChildStatus(&apiv1.PromotedChildStatus{Name: existingPromotedChildDef.GetName()})
-			} else {
-				rolloutObject.SetPromotedChildStatus(liveRolloutObject.GetPromotedChildStatus().DeepCopy())
-			}
```

Comment:
I relocated the setting of Promoted Child Status to the beginning of the `progressive.ProcessResource()` call - we should have the same promoted child throughout (up until the end of the Success case at least)